src/appShell/App/Container.tsx

```diff
@@ -85,12 +85,16 @@ export default class Container extends React.Component<IContainerProps, {}> {
     }
 
     @computed get isConfigComplete(){
-        // remote session will return undefined if there is no session id, so just check it
-        return configPromise.isComplete && this.routingStore.remoteSessionData.isComplete;
+
+        if (this.routingStore.needsRemoteSessionLookup) {
```

Comment:
basically, because were not bothering to check whether the session was in pending state, Container was causing the page component (injected by router) to unmount and remount unecessarily (because isConfigComplete would toggle between true/false)